### PR TITLE
feat(hierarchicalMenu): implement `getWidgetRenderState`

### PR DESCRIPTION
**Pull request by francoischalifour**
This implements the `getWidgetRenderState` widget lifecycle hook in `hierarchicalMenu`.

### CODE REVIEW

File: src/connectors/hierarchical-menu/connectHierarchicalMenu.js [hunk 1]
```diff
@@ -113,93 +113,45 @@ export default function connectHierarchicalMenu(renderFn, unmountFn = noop) {
...
-        // widget has requested more values (maxValuesPerFacet > getLimit()).
-        // Because this is used for making the search of facets unable or not, it is important
-        // to be conservative here.
-        const hasExhaustiveItems =
-          maxValuesPerFacetConfig > currentLimit
-            ? facetValues.length <= currentLimit
-            : facetValues.length < currentLimit;
+        const { renderState, instantSearchInstance } = renderOptions;
 
         this.toggleShowMore = this.createToggleShowMore(renderOptions);
```
Comment (Haroenv) [inline src/connectors/hierarchical-menu/connectHierarchicalMenu.js:148]: what is this function used for? It seems to be using `cachedToggleShowMore` in the other occurrences

File: src/connectors/configure/connectConfigure.ts [hunk 2]
```diff
@@ -99,25 +99,28 @@ const connectConfigure: ConfigureConnector = function connectConfigure(
     return {
       $$type: 'ais.configure',
 
-      init({ instantSearchInstance, helper }) {
+      init(initOptions) {
+        const { renderState, helper, instantSearchInstance } = initOptions;
```
Comment (Haroenv) [inline src/connectors/configure/connectConfigure.ts:103]: this seems to be from a different PR?

File: src/connectors/hierarchical-menu/connectHierarchicalMenu.js (hunk 1 above)
Comment (francoischalifour) [reply src/connectors/hierarchical-menu/connectHierarchicalMenu.js:148]: That's a proxy to `cachedToggleShowMore` that keeps the same function reference across renders. It was introduced a long time ago.

Comment (Haroenv) [reply src/connectors/hierarchical-menu/connectHierarchicalMenu.js:148]: I see now, thanks!
